fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/AvroDimensionRowParser.java

```diff
@@ -0,0 +1,192 @@
+        this.dimensionFieldMapper = memoize(dimensionFieldMapper);
+        this.objectMapper = objectMapper;
+    }
+
+    /**
+     * Validates the schema of the given AVRO file with the Dimension schema configured by the user.
+     *
+     * @param dimension The dimension object used to configure the dimension
+     * @param avroSchemaPath The path of the AVRO schema file (.avsc)
+     *
+     * @return true if the schema is valid, false otherwise
+     *
+     * @throws IllegalArgumentException thrown if JSON object `fields` is not present
+     * @throws IOException thrown if there is error parsing the avro files
+     *
+     * The sample schema is expected to be in the following format
+     *
+     * {
+     *    "type" : "record",
+     *    "name" : "TUPLE_0",
+     *    "fields" : [
+     *    {
+     *    "name" : "FOO_ID",
+     *    "type" : [ "null","int" ]
+     *    },
+     *    {
+     *    "name" : "FOO_DESC",
+     *    "type" : [ "null", "string" ]
+     *    }
+     *    ]
+     * }
+     *
+     */
+    public boolean isSchemaValid(Dimension dimension, String avroSchemaPath)
+        throws IllegalArgumentException, IOException {
+
+        // Convert the AVRO schema file into JsonNode Object
+        JsonNode jsonAvroSchema = convertFileToJsonNode(avroSchemaPath);
+
+        // If the JSON schema file is corrupted/ empty, then return the schema is invalid
+        if (jsonAvroSchema == null) {
+            return false;
+        }
+
+        try {
+            jsonAvroSchema = jsonAvroSchema.get("fields");
+        } catch (NullPointerException e) {
+            String msg = "`fields` is a required field in the avro schema {}";
+            LOG.error(msg, e);
+            throw new IllegalArgumentException(msg);
+        }
+
+        // Populating the set of avro field names
+        Set<String> avroFields = StreamSupport.stream(jsonAvroSchema.spliterator(), false)
+                .map(jsonNode -> jsonNode.get("name"))
+                .map(JsonNode::asText)
+                .collect(Collectors.toSet());
+
+        // True only if all of the mapped dimension fields are present in the Avro schema
+        return dimension.getDimensionFields().stream()
+                .map(dimensionField -> dimensionFieldMapper.convert(dimension, dimensionField))
+                .allMatch(avroFields::contains);
+    }
+
+    /**
+     * Parses the avro file and populates the dimension rows after validating the schema.
+     *
+     * @param dimension The dimension object used to configure the dimension
+     * @param avroFilePath The path of the AVRO data file (.avro)
+     * @param avroSchemaPath The path of the AVRO schema file (.avsc)
+     *
+     * @throws IllegalArgumentException thrown if JSON object `fields` is not present
+     * @throws IOException thrown if there is error parsing the avro files
+     */
+    public void parseAvroFileDimensionRows(Dimension dimension, String avroFilePath, String avroSchemaPath)
```

Comment:
Rather than adding the rows to the dimension in here, it's better to have this method _return_ the dimension rows it parses and let the caller decide what to do with them. Generally, we want to avoid side effects for methods when we can, and returning void is often an indication we are doing that. (Also, its odd for a parser to also add them to a dimension, instead of just returning the things it parsed)